lib/rubocop/cop/lint/safe_navigation_with_empty.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks to make sure safe navigation isn't used with `empty?` in
+      # a conditional.
+      #
+      # While the safe navigation operator is generally a good idea, when
+      # checking `foo&.empty?` in a conditional, `foo` being `nil` will actually
+      # do the opposite of what the author intends.
+      #
+      # @example
+      #   # bad
+      #   return if foo&.empty?
+      #   return unless foo&.empty?
```

Comment:
What about `blank?`. `foo.blank?` does not need safe navigation operator, so `return if foo&.blank?` is not only incorrect because the problem described here, but it is also completely unnecessary to use safe navigation operator because `nil` also responds to `blank?`. This is Rails-specific, though, do we have a different repository with Rails-specific cops? :thinking:

Reply:
> This is Rails-specific, though, do we have a different repository with Rails-specific cops? 🤔

We have extracted `rubocop-rails`, but the Rails specific cops haven't been moved there yet. But you are correct that it shouldn't go in this cop. 🙂

Reply:
Even once we are done extracting `rubocop-rails`, that project won't be able to change the behavior of this cop without redefining functionality. I think it could be reasonable to have a configuration setting to optionally use `blank?`.